tests/contract/test_resource_client.py

```diff
@@ -670,3 +685,28 @@ def test_is_primary_identifier_equal_fail(resource_client):
         {"a": 1, "b": 2, "c": 3},
         {"a": 1, "b": 2, "c": 4},
     )
+
+
+def test_assert_write_only_property_does_not_exist(resource_client):
+    schema = {
+        "a": {"type": "number", "const": 1},
+        "b": {"type": "number", "const": 2},
+        "c": {"type": "number", "const": 3},
+    }
+    resource_client._update_schema(schema)
+    resource_client.assert_write_only_property_does_not_exist(schema)
+
+
+@pytest.mark.parametrize("schema", [SCHEMA, SCHEMA_WITH_MULTIPLE_WRITE_PROPERTIES])
+def test_assert_write_only_property_does_not_exist_success(resource_client, schema):
+    created_resource = {"a": None, "b": 2, "c": 3, "d": None}
```

Comment:
what happens if a and b arent specified at all? it should handle it in this case

Reply:
checking for that now.